src/Lykke.Job.TransactionHandler/Handlers/EthereumCoreCommandHandler.cs

```diff
@@ -0,0 +1,468 @@
+        {
+            CashOutContextData context = await _transactionService.GetTransactionContext<CashOutContextData>(queueMessage.OperationId);
+
+            if (context != null)
+            {
+                string transactionHandlerUserId = "auto redeem";
+                string clientId = context.ClientId;
+                string hash = queueMessage.TransactionHash;
+                string cashOperationId = context.CashOperationId;
+                string assetId = context.AssetId;
+                var amount = context.Amount;
+
+                try
+                {
+                    var asset = await _assetsService.AssetGetAsync(assetId);
+                    await _cashOperationsRepositoryClient.UpdateBlockchainHashAsync(clientId, cashOperationId, hash);
+                    var pt = await _paymentTransactionsRepository.TryCreateAsync(PaymentTransaction.Create(hash,
+                                CashInPaymentSystem.Ethereum, clientId, (double)amount,
+                                asset.DisplayId ?? asset.Id, status: PaymentStatus.Processing));
+                    if (pt == null)
+                    {
+                        await
+                            _log.WriteWarningAsync(nameof(EthereumCoreCommandHandler), nameof(ProcessFailedCashout), hash,
+                                "Transaction already handled");
+                        //return if was handled previously
+                        return false;
+                    }
+
+                    var sign = "+";
+                    var commentText =
+                        $"Balance Update: {sign}{amount} {asset.Name}. Cashout failed: {hash} {queueMessage.OperationId}";
+
+                    var newComment = new ClientComment
+                    {
+                        ClientId = clientId,
+                        Comment = commentText,
+                        CreatedAt = DateTime.UtcNow,
+                        FullName = "Lykke.Job.TransactionHandler",
+                        UserId = transactionHandlerUserId
+                    };
+
+                    var exResult = await _exchangeOperationsServiceClient.ManualCashInAsync(clientId, assetId,
+                            (double)amount, "auto redeem", commentText);
+
+                    if (!exResult.IsOk())
+                    {
+                        await
+                            _log.WriteWarningAsync(nameof(EthereumCoreCommandHandler), nameof(ProcessFailedCashout),
+                            (new
+                            {
+                                ExchangeServiceResponse = exResult,
+                                QueueMessage = queueMessage
+                            }).ToJson(),
+                                "ME operation failed");
+
+                        return false;
+                    }
+
+                    await _clientCommentsRepository.AddClientCommentAsync(newComment);
```

Comment:
Add ChaosKitty before and after this call

Reply:
fixed